WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -623,6 +623,17 @@ private extension OrderDetailsViewController {
         }
     }
 
+    private func retryCollectPayment() {
+        self.viewModel.cancelPayment { [weak self] result in
```

Comment:
We cancel the previous payment and start a new one from scratch. 

If the cancellation of the previous payment fails, we just present a modal error that does not allow a retry. I can't think of a better way to handle this, retrying again is a possibility if users tap the "collect payment" button